components/server/src/workspace/workspace-starter.ts

```diff
@@ -1257,6 +1256,10 @@ export class WorkspaceStarter {
 
             const result = await client.build({ span }, req, imageBuildLogInfo);
 
+            if (result.actuallyNeedsBuild) {
+                increaseImageBuildsStartedTotal();
```

Comment:
At what point does this increment? Is it after we've completed the build? Or is the `build` method async and it tells us it needs to be build and returns?

Reply:
the `build` call on line 1257 awaits until the build *starts*:

https://github.com/gitpod-io/gitpod/blob/c551c2ca79eefcdbf3778a254a3abc507230f175/components/image-builder-api/typescript/src/sugar.ts#L176-L177

so we're incrementing on build start here, not completion.

The build completion is awaited further down this file on line 1281:

https://github.com/gitpod-io/gitpod/blob/c551c2ca79eefcdbf3778a254a3abc507230f175/components/server/src/workspace/workspace-starter.ts#L1281

Reply:
Thanks!